### PR TITLE
adding cloudprovider

### PR DESCRIPTION
**Pull request by puja108**
towards https://github.com/giantswarm/giantswarm/issues/1588

needed for https://github.com/giantswarm/k8scloudconfig/pull/157

### CODE REVIEW

**Review by rossf7 — APPROVED** [review]
LGTM. I think it's better to add this to clustertpr than add conditional logic to k8scloudconfig.

**Comment by rossf7** [comment]
Ping @xh3b4sd @kopiczko for info. This is so we can use the dynamic storage provider on AWS. The cloud provider value is needed for apiserver.

**Comment by puja108** [comment]
actually for api-server, controller-manager, and kubelets and we will need this flexible to allow for other cloud providers in the future.

File: kubernetes/kubernetes.go [hunk 1]
```diff
@@ -17,6 +17,10 @@ type Kubernetes struct {
 	// Domain is the base domain of the Kubernetes cluster, e.g.
 	// g8s.fra-1.giantswarm.io.
 	Domain            string                    `json:"domain" yaml:"domain"`
+	// CloudProvider enables cloud provider specific functionality
+	// can be aws, azure, gce, ... needs to be unset for baremetal
+	// see https://kubernetes.io/docs/getting-started-guides/scratch/#cloud-providers)
+	CloudProvider     string                    `json:"cloudProvider" yaml:"cloudProvider"`
```
Comment (xh3b4sd) [inline kubernetes/kubernetes.go:23]: This breaks the alphabetical order.

Comment (puja108) [reply kubernetes/kubernetes.go:23]: oh, i didn't see that there was an alphabetical order, I can issue a fix

Comment (xh3b4sd) [reply kubernetes/kubernetes.go:23]: It is not mission critical. I just noticed it.

**Pull request merged**